app/javascript/geoblacklight/downloaders/downloader.js

```diff
@@ -0,0 +1,62 @@
+export default class GeoBlacklightDownloader {
+  constructor(el, options) {
+    this.el = el;
+    this.options = options || {};
+    this.options.spinner = document.createElement("div");
+    this.options.spinner.className =
+      "spinner-border spinner-border-sm float-right";
+    this.options.spinner.role = "status";
+    this.options.spinner.innerHTML = '<span class="sr-only">Downloading</span>';
+    this.options.spinner.style.display = "none";
+    // @TODO: This spinner doesn't appear to work in GBL Proper anymore.
+    //document.querySelector('.exports .card-header').appendChild(this.options.spinner);
+    this.configureHandler();
+  }
+
+  configureHandler() {
+    this.el.addEventListener("click", (ev) => this.download(ev));
+  }
+
+  download(ev) {
+    ev.preventDefault();
+    if (this.downloading) {
+      return;
+    }
+    this.downloading = true;
+    const url = this.el.getAttribute("data-download-path");
+    this.options.spinner.style.display = "block";
+
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+        return response.json();
+      })
+      .then((data) => this.complete(data))
+      .catch((error) => this.error(error));
+  }
+
+  complete(data) {
+    this.downloading = false;
+    this.el.disabled = false;
```

Comment:
This is not in the old code but it should be, download should have a line that says this.el.disabled = true.

Reply:
Actually after investigations since these are links, disabled doesn't do anything. We need to create a disabled class for links for this to work.